### PR TITLE
ENH: Add commit message github actions workflow

### PR DESCRIPTION
**Pull request by jamesobutler**
This replaces the usage of https://github.com/marketplace/commitcheck which has been unreliable at times and also difficult to see what is required to pass a failing check. I had to find https://discourse.slicer.org/t/commit-message-prefix/10921 to see the previously used regex. The unreliable can be observed in recent PR #6601 where the installed CommitCheck app is not responding and the https://www.commitcheck.com/ site is unresponsive.

image

This check is based on content at https://slicer.readthedocs.io/en/latest/developer_guide/style_guide.html#commits.

### CODE REVIEW

File: .github/workflows/commit-message.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check Commit Prefix
+        uses: gsactions/commit-message-checker@v2
+        with:
+          pattern: '^(ENH|PERF|BUG|STYLE|DOC|COMP): ([A-Z])+'
+          flags: 'gm'
+          excludeDescription: 'true' # optional: this excludes the description body of a pull request
+          excludeTitle: 'true' # optional: this excludes the title of a pull request
+          error: 'Your first line has to contain a commit type like "ENH: Add something".'
```
Comment (jcfr) [inline .github/workflows/commit-message.yml:25]:
```suggestion
          error: 'The first line (called the commit title) must include the type (e.g   "ENH: Add something"). For more details, see https://slicer.readthedocs.io/en/latest/developer_guide/style_guide.html#commits.'

**Comment by jcfr** [comment]
I will disable `CommitCheck App` from the following repositories:

image

**Comment by jcfr** [comment]
Example of output for prefix check:

image

**Comment by jamesobutler** [comment]
@jcfr Looks like your commit had wrapping of the subject line? The check failed for my really long commit subject

**Comment by jcfr** [comment]
> Looks like your commit had wrapping of the subject line

Good catch. Look like my git client outsmarted me. I will cleanup the PR

**Comment by jcfr** [comment]
Summary
* CommitCheck has been removed from the `Slicer` repository
* Since the current hook have been enforcing 78 characters since 2010, the documentation and workflow have been updated accordingly

**Pull request merged**